dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java

```diff
@@ -145,8 +145,13 @@ private void deleteUnrecognizedPage(Path path) {
   private Optional<PageId> getPageId(Path path) {
     Matcher matcher = mPagePattern.matcher(path.toString());
     if (!matcher.matches()) {
-      // Please note, TEMP page files are also treated as Unrecognized!!
-      LOG.error("Unrecognized page file " + path + ". Let's delete it.");
+      if (Pattern.matches(String.format("%s/%d/%s/([^/]+)/(\\d+)",
+          Pattern.quote(mPageStoreOptions.getRootDir().toString()),
```

Comment:
this will compile the same regex pattern every time instead of reusing it. we can cache the regex pattern object to save that cost.

Reply:
Do you like this patch, @dbw9580 ?
```
diff --git a/dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java b/dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java
index 599fb2c61a..098eed712b 100644
--- a/dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java
+++ b/dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java
@@ -45,6 +45,8 @@ public class LocalPageStoreDir extends QuotaManagedPageStoreDir {
 
   private PageStore mPageStore;
 
+  private Pattern mPatternForTempPage = null;
+
   /**
    * Constructor for LocalCacheDir.
    * @param pageStoreOptions
@@ -95,8 +97,13 @@ public class LocalPageStoreDir extends QuotaManagedPageStoreDir {
    */
   @Override
   public void scanPages(Consumer<Optional<PageInfo>> pageInfoConsumer) throws IOException {
+    mPatternForTempPage = Pattern.compile(String.format("%s/%d/%s/([^/]+)/(\\d+)",
+            Pattern.quote(mPageStoreOptions.getRootDir().toString()),
+            mPageStoreOptions.getPageSize(), LocalPageStore.TEMP_DIR));
+
     Files.walk(getRootPath()).filter(Files::isRegularFile).map(this::getPageInfo)
         .forEach(pageInfoConsumer);
+    mPatternForTempPage = null;
   }
 
   /**
@@ -145,9 +152,8 @@ public class LocalPageStoreDir extends QuotaManagedPageStoreDir {
   private Optional<PageId> getPageId(Path path) {
     Matcher matcher = mPagePattern.matcher(path.toString());
     if (!matcher.matches()) {
-      if (Pattern.matches(String.format("%s/%d/%s/([^/]+)/(\\d+)",
-          Pattern.quote(mPageStoreOptions.getRootDir().toString()),
-          mPageStoreOptions.getPageSize(), LocalPageStore.TEMP_DIR), path.toString())) {
+      Matcher matcherTemp = mPatternForTempPage.matcher(path.toString());
+      if (matcherTemp.matches()) {
         LOG.info("TEMP page file " + path + " is going to be deleted.");
       } else {
         LOG.error("Unrecognized page file " + path + "is going to be deleted.");

```

Personally I don't think it is worthwhile to do so.

Reply:
> this is only used at startup time, and I don't want this object to be there, occupying some memory all the time.

Right, but this is called for every page, so it will slow down the startup procedure a lot. Imagine you have multiple millions of pages in the storage directory. Plus, it creates an object every time `Pattern.matches` is called, so the memory usage is certainly a lot more than if it is a static class member, which is a singleton and takes a tiny amount of memory, albeit throughout the whole time the process is running.

Reply:
As discussed with @dbw9580 , we will create a new task to optimize the regex pattern here.